### PR TITLE
feat: Add Sparkplug B plugin with device-level PARRIS architecture

### PR DESCRIPTION
**Pull request by JeremyTheocharis**
- Add Sparkplug B Input plugin (Primary Host role) with MQTT session lifecycle management
- Add Sparkplug B Output plugin (Edge Node role) with device-level PARRIS implementation
- Add Sparkplug B Decode processor for protobuf message processing
- Implement static Edge Node ID + dynamic Device ID from location_path for Sparkplug B compliance
- Add comprehensive test suite with 89 unit tests and integration testing
- Add complete documentation for input, output, and processor plugins
- Support all Sparkplug B message types: NBIRTH, DBIRTH, NDATA, DDATA, NDEATH, DDEATH
- Include alias management, sequence validation, and proper message lifecycle
- Industry-aligned architecture matching Ignition MQTT Transmission patterns

Features:
- Full Sparkplug B v3.0 specification compliance
- Device-level PARRIS method for dynamic device identification
- Automatic alias resolution and metric name mapping
- MQTT session management with rebirth requests
- Complete protobuf encoding/decoding
- Comprehensive error handling and logging
- Integration with UMH tag processor and unified namespace

Testing:
- 89 unit tests covering all components and edge cases
- Integration tests with live MQTT broker communication
- Performance validation and memory usage optimization
- Real-world Sparkplug B message flow validation

### CODE REVIEW

File: sparkplug_plugin/sparkplug_b_input.go [hunk 1]
```diff
@@ -0,0 +1,1179 @@
...
+	}
+
+	// Create SparkplugMessage struct for conversion
+	sparkplugMsg := &SparkplugMessage{
+		GroupID:    topicInfo.Group,
+		EdgeNodeID: topicInfo.EdgeNode,
+		DeviceID:   topicInfo.Device,
+		Value:      rawValue,
+		DataType:   s.convertSparkplugDataTypeToString(*metric.Datatype),
+		Timestamp:  time.Now(), // Will be overridden below if payload has timestamp
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sparkplug_plugin/sparkplug_b_input.go]: Fixed in commit fa481e3: Added nil check before dereferencing metric.Datatype in tryAddUMHMetadata function (line 1088-1095). Now safely handles metrics without datatype field.

File: sparkplug_plugin/unit_test.go [hunk 2]
```diff
@@ -0,0 +1,2660 @@
...
+				uint64(42):    "uint64",
+				float32(3.14): "float",
+				float64(3.14): "double",
+				"hello":       "string",
+			}
+
+			for value, expectedType := range values {
+				inferredType := converter.InferMetricType(value)
+				Expect(inferredType).To(Equal(expectedType), "Value %v should infer type %s", value, expectedType)
+			}
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sparkplug_plugin/unit_test.go]: Fixed in commit fa481e3: Corrected Gomega assertion formatting using fmt.Sprintf instead of relying on Gomega's variadic arguments. The placeholders are now properly interpolated.

File: sparkplug_plugin/unit_test.go [hunk 3]
```diff
@@ -0,0 +1,2660 @@
...
+					Alias:    uint64Ptr(1),
+					Datatype: uint32Ptr(7), // Int64
+					Value:    &sparkplugb.Payload_Metric_LongValue{LongValue: 12345},
+				},
+				{
+					Name:     stringPtr("Temperature"),
+					Alias:    uint64Ptr(100),
+					Datatype: uint32Ptr(9), // Float
+					Value:    &sparkplugb.Payload_Metric_FloatValue{FloatValue: 25.5},
+				},
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sparkplug_plugin/unit_test.go]: Fixed in commit fa481e3: Changed bdSeq datatype from Int64 (7) to UInt64 (8) in unit_test.go line 939 to comply with Sparkplug B v3.0 specification. The bdSeq field must be an unsigned 64-bit integer per spec.

File: sparkplug_plugin/unit_test.go [hunk 4]
```diff
@@ -0,0 +1,2660 @@
...
+			}
+
+			// Cache all metrics
+			start := time.Now()
+			count := cache.CacheAliases("Factory/LargeLine", largeMetrics)
+			duration := time.Since(start)
+
+			Expect(count).To(Equal(500), "Should cache all 500 metrics")
+			Expect(duration).To(BeNumerically("<", 100*time.Millisecond), "Should cache quickly")
+		})
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sparkplug_plugin/unit_test.go:1452]: Fixed in commit fa481e3: Increased time-based thresholds throughout integration tests to prevent CI flakes. Sleep durations increased from 100ms to 200-500ms, and context timeouts increased from 10-20s to 20-30s for better stability in slower CI environments.
